homeassistant/components/eq3btsmart/__init__.py

```diff
@@ -0,0 +1,153 @@
+"""Support for EQ3 devices."""
+
+import logging
+from typing import Any
+
+from bleak.backends.device import BLEDevice
+from bleak_esphome.backend.scanner import ESPHomeScanner
+from eq3btsmart import Thermostat
+from eq3btsmart.thermostat_config import ThermostatConfig
+
+from homeassistant.components import bluetooth
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MAC, CONF_NAME, CONF_SCAN_INTERVAL, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import (
+    CONF_ADAPTER,
+    CONF_CURRENT_TEMP_SELECTOR,
+    CONF_EXTERNAL_TEMP_SENSOR,
+    CONF_TARGET_TEMP_SELECTOR,
+    DEFAULT_ADAPTER,
+    DEFAULT_CURRENT_TEMP_SELECTOR,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_TARGET_TEMP_SELECTOR,
+    DOMAIN,
+    Adapter,
+)
+from .models import Eq3Config, Eq3ConfigEntry
+
+PLATFORMS = [
+    Platform.CLIMATE,
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Handle config entry setup."""
+
+    mac_address: str = entry.data[CONF_MAC]
+    name: str = entry.data[CONF_NAME]
+    adapter: Adapter = entry.options.get(CONF_ADAPTER, DEFAULT_ADAPTER)
+    current_temp_selector = entry.options.get(
+        CONF_CURRENT_TEMP_SELECTOR, DEFAULT_CURRENT_TEMP_SELECTOR
+    )
+    target_temp_selector = entry.options.get(
+        CONF_TARGET_TEMP_SELECTOR, DEFAULT_TARGET_TEMP_SELECTOR
+    )
+    external_temp_sensor = entry.options.get(CONF_EXTERNAL_TEMP_SENSOR, "")
+    scan_interval = entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
```

Comment:
those are currently empty then, aren't they?

Reply:
Yep you're right, will remove.